### PR TITLE
Flutter SDK setup in Dockerfile

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,9 +4,17 @@ FROM google/dart-runtime-base:1.23.0
 # docker image diff small.
 RUN apt-get update && \
     apt-get upgrade -y && \
-    apt-get install -y memcached && \
+    apt-get install -y git memcached unzip && \
     rm -rf /var/lib/apt/lists/*
 
+ENV PATH="/flutter/bin:${PATH}"
+# To ensure actions from this container are not seen as "typical" users.
+ENV PUB_ENVIRONMENT="bot.pub_dartlang_org.docker"
```
Comment (isoos) [inline Dockerfile]: /cc @kevmoo

Comment (kevmoo) [inline Dockerfile]: 👍

File: Dockerfile [hunk 2]
```diff
@@ -4,9 +4,17 @@ FROM google/dart-runtime-base:1.23.0
...
+
+RUN git clone -b alpha https://github.com/flutter/flutter.git
```
Comment (mkustermann) [inline Dockerfile]: Using a branch might make this a non-reproducible docker build. Maybe use the tag '0.0.7' instead (to which the 'alpha' branch is currently set)

File: Dockerfile [hunk 3]
```diff
@@ -4,9 +4,17 @@ FROM google/dart-runtime-base:1.23.0
 # docker image diff small.
 RUN apt-get update && \
     apt-get upgrade -y && \
-    apt-get install -y memcached && \
+    apt-get install -y git memcached unzip && \
     rm -rf /var/lib/apt/lists/*
 
+ENV PATH="/flutter/bin:${PATH}"
+# To ensure actions from this container are not seen as "typical" users.
```
Comment (mkustermann) [inline Dockerfile]: Maybe rephrase this a bit:

`Let the pub server know that this is not a "typical" pub client but rather a bot.`

File: Dockerfile [hunk 4]
```diff
@@ -4,9 +4,17 @@ FROM google/dart-runtime-base:1.23.0
...
+ENV PUB_ENVIRONMENT="bot.pub_dartlang_org.docker"
+
+RUN git clone -b alpha https://github.com/flutter/flutter.git
+# Downloads the Dart SDK and disables analytics tracking – which we always want
+RUN flutter config --no-analytics
```
Comment (mkustermann) [inline Dockerfile]: Maybe add a warning comment here that this will add 400 MB to our custom docker layer

File: Dockerfile [hunk 5]
```diff
@@ -4,9 +4,17 @@ FROM google/dart-runtime-base:1.23.0
 # docker image diff small.
```
Comment (mkustermann) [inline Dockerfile]: Maybe change the `FROM ...` to upgrade to the latest stable version, namely 1.24.0.

File: Dockerfile (hunk 2 above)
Comment (isoos) [inline Dockerfile]: Please check this comment: https://github.com/dart-lang/pana/issues/17#issuecomment-306454385

I don't mind fixing it to a specific version.

File: Dockerfile (hunk 4 above)
Comment (isoos) [inline Dockerfile]: Alternatively, we could do most of these from the analyzer service, on its initialization. That way this extra 400MB will be on that instance only, and not on `frontend` or on soon-to-be `search`.

**Comment by kevmoo** [comment]
+1

We could have a script that manages this on boot. One concern: I'm not sure
if we have to worry about reboot...

**Comment by isoos** [comment]
Closing this in favor of a boot-time setup.
